src/datatypes/src/value.rs

```diff
@@ -441,6 +441,56 @@ impl Ord for Value {
     }
 }
 
+macro_rules! impl_try_from_value {
+    ($Variant: ident, $Type: ident) => {
+        impl TryFrom<Value> for $Type {
+            type Error = ();
+
+            #[inline]
+            fn try_from(from: Value) -> std::result::Result<Self, Self::Error> {
+                match from {
+                    Value::$Variant(v) => Ok(v.into()),
+                    _ => Err(()),
+                }
+            }
+        }
+
+        impl TryFrom<Value> for Option<$Type> {
+            type Error = ();
```

Comment:
Could we define an error variant for this case?
```suggestion
            type Error = Error;

            #[inline]
            fn try_from(from: Value) -> std::result::Result<Self, Self::Error> {
                match from {
                    Value::$Variant(v) => Ok(v.into()),
                    _ => FromValue {
                        reason: format!("{:?} is not a {}", from, stringify!($Type)),
                    }.fail(),
                }
            }
        }

        impl TryFrom<Value> for Option<$Type> {
            type Error = ();
```